### DISCUSSION TITLE
Two simultaneous I2S modules in TX mode possible?

### DISCUSSION THREAD

**Discussion by TodorovS** [pschatzmann/arduino-audio-tools#584] [Q&A] [answered]
Hello,
I want to build a 2.1 system (two speakers, one subwoofer), with Bluetooth and Internet radio option.
In the ESP32 documentation is marked, that there are two independent i2s modules I2S_0 and I2S_1, which can be set either in TX and RX mode.

My idea is to use BT or Internet radio as a source, then sink the data to both I2S streams.
One of them will be output to external DAC straight away for the L and R speakers.
A low pass filter/DSP will be applied to the other one and then sunk to a separate DAC which feeds the subwoofer amplifier.

Do you think it is possible to I initialize two I2S streams (utilizing I2S_0 and I2S_1) and config two DataOut pins?
If yes, can you give an example code how to do it?
Or if it is not possible can I somehow workaround this, to spit the stream for example or whatsoever?

Any help will be highly appreciated!
Kind regads!

**Comment by pschatzmann** [comment, marked as answer]
The following [exemple](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-i2s-i2s-2/streams-i2s-i2s-2.ino) uses the 2 I2S channels. Just create a MultiOutput and add the to streams to it...

I have never done something like this, so I do not know about the pitfalls....

**Reply by TodorovS** [reply]
Thanks for Your prompt reply.
So in theory, from the example You gave, I can define two I2SStreams in TX mode

I2SStream out_mains;
I2Sstream out_subwoofer;

and then config both streams:
auto config_out_mains =out_mains.defaultConfig(TX_MODE); 
with its dedicated BCK, WS and Data pins.

Then do the same for out_subwoofer?

I just don't comprehend the basic workflow.
As far as I get it, I should set up a basic  BT receiver with a2dp sink (from the your ESP32-a2dp library). The a2dp stream should be "duplicated" with MultiOutput (output two identical a2dp streams). I can then utilize a2dp-i2s example and "attach the two a2dp streams to out_mains and out_subwoofer"

Is that correct?

**Comment by pschatzmann** [comment]
Take [this example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-i2s/basic-a2dp-i2s.ino) and instead of writing the data to the I2SStream you would write the a MultiOutput which consists of 2 I2SStream.  When setting up the I2S also make sure that you a different port for each.
